internal/promclient/promclient.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package promclient
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/omec-project/metricfunc/config"
+	"github.com/omec-project/metricfunc/logger"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type PromStats struct {
+	coreSub     *prometheus.CounterVec
+	smfSessions *prometheus.GaugeVec
+	nfStatus    *prometheus.GaugeVec
+}
+
+var promStats *PromStats
+
+func init() {
+	promStats = initPromStats()
+
+	if err := promStats.register(); err != nil {
+		log.Panicln("SMF Stats register failed ", err.Error())
+	}
+}
+
+func StartPrometheusClient(cfg *config.ServerAddr) {
+	logger.PromLog.Debugf("prometheus server initialised on address [%v] port [%v] ", cfg.Addr, cfg.Port)
+	HTTPAddr := fmt.Sprintf(":%d", cfg.Port)
+	http.Handle("/metrics", promhttp.Handler())
+	http.ListenAndServe(HTTPAddr, nil)
+}
+
+func initPromStats() *PromStats {
+	return &PromStats{
+		coreSub: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "core_subscriber",
+			Help: "core subscriber info",
+		}, []string{"imsi", "ip_addr", "state", "smf_ip", "dnn", "slice", "upf"}),
+
+		smfSessions: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "smf_pdu_sessions",
+			Help: "Number of SMF PDU sessions currently in the SMF",
```

Comment:
Should be - "Current Number of PDU sessions in Core"

Reply:
Right, fixed it